spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/token/flows/ClientCredentialsTokenFlow.java

```diff
@@ -0,0 +1,238 @@
+	 * The secret needs to be the one of the client that requests the token.
+	 * 
+	 * @param clientSecret
+	 *            - the secret of the OAuth 2.0 client requesting the token.
+	 * @return this builder.
+	 */
+	public ClientCredentialsTokenFlow secret(String clientSecret) {
+		request.setClientSecret(clientSecret);
+		return this;
+	}
+
+	/**
+	 * Adds additional authorization attributes to the request. <br>
+	 * Clients can use this to request additional attributes in the
+	 * {@code 'az_attr'} claim of the returned token.
+	 * 
+	 * @param additionalAuthorizationAttributes
+	 *            - the additional attributes.
+	 * @return this builder.
+	 */
+	public ClientCredentialsTokenFlow attributes(Map<String, String> additionalAuthorizationAttributes) {
+		this.request.setAdditionalAuthorizationAttributes(additionalAuthorizationAttributes);
+		return this;
+	}
+
+	/**
+	 * Executes the token flow and returns a JWT token from XSUAA.
+	 * 
+	 * @return the JWT token generated by XSUAA.
+	 * @throws TokenFlowException
+	 *             in case of token flow errors.
+	 */
+	public Jwt execute() throws TokenFlowException {
+
+		checkRequest(request);
+
+		return requestTechnicalUserToken(request);
+	}
+
+	/**
+	 * Checks if the built request is valid. Throws an exception if not all
+	 * mandatory fields are filled.
+	 * 
+	 * @param request
+	 *            - the token flow request.
+	 * @throws TokenFlowException
+	 *             in case the request does not have all mandatory fields set.
+	 */
+	private void checkRequest(XsuaaTokenFlowRequest request) throws TokenFlowException {
+		if (!request.isValid()) {
+			throw new TokenFlowException(
+					"Client credentials flow request is not valid. Make sure all mandatory fields are set.");
+		}
+	}
+
+	/**
+	 * Requests the client credentials token from XSUAA.
+	 * 
+	 * @param request
+	 *            - the token request.
+	 * @return the JWT token returned by XSUAA.
+	 * @throws TokenFlowException
+	 *             in case of an error during the flow.
+	 */
+	private Jwt requestTechnicalUserToken(XsuaaTokenFlowRequest request) throws TokenFlowException {
+
+		UriComponentsBuilder builder = UriComponentsBuilder.fromUri(request.getTokenEndpoint());
+
+		// add grant type to URI
+		builder.queryParam(GRANT_TYPE, CLIENT_CREDENTIALS);
+
+		String authorities = buildAuthorities(request); // returns JSON!
+		if (authorities != null) {
+			builder.queryParam(AUTHORITIES, authorities); // places JSON inside the URI !?!
+		}
+
+		HttpHeaders headers = createHeadersForTechnicalUserTokenExchange(request);
+
+		HttpEntity<Void> requestEntity = new HttpEntity<>(headers);
+
+		URI requestUri = builder.build().encode().toUri();
+
+		@SuppressWarnings("rawtypes")
+		ResponseEntity<Map> responseEntity = restTemplate.postForEntity(requestUri, requestEntity, Map.class);
+
+		HttpStatus responseStatusCode = responseEntity.getStatusCode();
+
+		if (responseStatusCode == HttpStatus.UNAUTHORIZED) {
```

Comment:
Wrong error handling: if you like to handle erronous calls you need to catch 
org.springframework.web.client.HttpClientErrorException